### PR TITLE
Version 0.1.5

### LINKED ISSUE #2: Allow for cache size to be configurable
Currently, the [lrucache used by this module](https://github.com/openresty/lua-resty-lrucache) is [hardcoded to hold up to 200 items](https://github.com/mozilla-services/iprepd-nginx/blob/master/lib/resty/iprepd.lua#L21)

Should allow for this to be configurable with a good default + add documentation around this setting.

### LINKED ISSUE #10: Add support for injecting headers
```
X-Foxsec-IP-Reputation: <int>
X-Foxsec-IP-Reputation-Below-Threshold: "true"/"false"
X-Foxsec-Block: "true"/"false"
```

### LINKED ISSUE #14: Make statsd flush timer configurable

### LINKED ISSUE #15: Support whitelisting in iprepd-nginx
Support whitelisting with cidr notation supported.

### LINKED ISSUE #16: Support log-only option
Support option where it just logs/send stats for accepted/rejected rather than actually blocking.

### PR DESCRIPTION
**Pull request by ajvb**
- Adds support for a "dont block" mode. Fixes #16
- Adds injecting headers by default. Fixes #10
- Makes cache size configurable. Fixes #2
- Adds support for an IP whitelist. Fixes #15
- Makes statsd flush timer configurable. Fixes #14 

cc @milescrabill

### CODE REVIEW

File: lib/resty/iprepd.lua [hunk 1]
```diff
@@ -34,33 +34,73 @@ function _M.new(options)
...
+    if reputation <= self.threshold then
+      ngx.req.set_header('X-Foxsec-IP-Reputation-Below-Threshold', 'true')
+      ngx.req.set_header('X-Foxsec-Block', 'true')
+      if self.statsd then
+        self.statsd.incr("iprepd.status.rejected")
+      end
 
-  -- Get reputation for ip
+      if self.dont_block == 1 then
+        ngx.log(ngx.ERR, '[logonly] ' .. ip .. ' rejected with a reputation of ' .. reputation)
```
Comment (ameihm0912) [inline lib/resty/iprepd.lua]: Should we return after this line, so the subsequent header sets (for below-threshold to false after this code block) are not executed?

**Review by ameihm0912 — APPROVED** [review]

File: lib/resty/iprepd.lua (hunk 1 above)
Comment (ajvb) [reply lib/resty/iprepd.lua]: At the moment, I'm thinking that we should always set the headers. I think I should change this log message to be a little more explicit. As well, should change the name of `self.statsd.incr("iprepd.status.rejected")`

Comment (ameihm0912) [reply lib/resty/iprepd.lua]: 👍

Comment (ajvb) [reply lib/resty/iprepd.lua]: Oh haha, I need to drink more coffee before, you are completely right. I misunderstood the comment.

**Pull request merged**